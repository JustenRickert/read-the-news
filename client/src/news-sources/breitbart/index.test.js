const puppeteer = require('puppeteer')

const { collectArticle } = require('../')

const TIMEOUT = 30e3

describe('breitbart snapshots', () => {
  jest.setTimeout(TIMEOUT)

  let browser
  let page

  beforeAll(async () => {
    browser = await puppeteer.launch()
  })

  beforeEach(async () => {
    page = await browser.newPage()
  })

  afterEach(async () => {
    await page.close()
  })

  afterAll(async () => {
    await browser.close()
  })

  it('does articles without subheadings', async () => {
    const result = await collectArticle(
      page,
      'https://www.breitbart.com/clips/2019/10/11/msnbcs-wallace-on-shep-smiths-fox-news-departure-very-sad-news-for-the-truth-today/'
    )
    expect(result).toMatchSnapshot()
  })

  it('does not do ending citations', async () => {
    const result = await collectArticle(
      page,
      'https://www.breitbart.com/politics/2019/11/21/watch-david-holmes-refuses-to-say-he-will-be-more-careful-with-communications/'
    )
    expect(result).toMatchSnapshot()
  })

  it('does articles with weird endings', async () => {
    const result = await collectArticle(
      page,
      'https://www.breitbart.com/politics/2019/08/05/kenny-marchant-fourth-texas-republican-not-seek-re-election/'
    )
    expect(result).toMatchSnapshot()
  })

  it('does not put the subheading in the article and correctly ignore ad', async () => {
    const result = await collectArticle(
      page,
      'https://www.breitbart.com/europe/2019/11/23/court-dutch-govt-does-not-have-repatriate-children-syria/'
    )
    expect(result).toMatchSnapshot()
  })

  it('ignores twitter widgets and puts smart quotes around blockquotes', async () => {
    const result = await collectArticle(
      page,
      'https://www.breitbart.com/border/2019/11/05/narco-terror-10-u-s-women-children-murdered-by-cartel-gunmen-near-new-mexico-border/'
    )
    expect(result).toMatchSnapshot()
  })
})
