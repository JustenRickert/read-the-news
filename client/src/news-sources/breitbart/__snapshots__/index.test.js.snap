// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breitbart snapshots does articles with weird endings 1`] = `
Object {
  "authors": Array [
    Object {
      "href": "https://www.breitbart.com/author/hannah-bleau/",
      "name": "Hannah Bleau",
    },
  ],
  "content": "While Marchant said he will finish his term, he stated that he is looking forward to “returning to Texas to start a new chapter.”
“It is time for me to announce that I will not seek another term as Congressman from the 24th District of Texas,” Marchant announced Monday. “I am looking forward to finishing out my term and then returning to Texas to start a new chapter.”
Marchant, who was first elected in 2004, won the 24th House District by a relatively narrow margin in 2018– 50.6 percent to Democrat challenger Jan McDowell’s 47.5 percent. The district has been described as “highly competitive”:
His announcement comes on the heels of announcements from Reps. Pete Olson, Will Hurd, and K. Michael Conaway, all of whom said they will not seek reelection.
Like Marchant, Hurd narrowly beat his Democrat challenger in 2018, defeating Gina Ortiz Jones by less than one percent. The departures have sparked concern from Republicans, who fear a coming blue wave in the red state. Rep. Alexandria Ocasio-Cortez (D-NY) predicted last week that Democrats are “going to work hard and win in Texas”:
Democrats reportedly spent $138 million on the last election cycle, which assisted them in taking coveted House seats. As Breitbart News reported, the GOP has been beefing up resources in Texas in hopes of preventing a blue wave on both congressional and executive levels:
“Donors are reportedly strengthening efforts in the state and primarily focusing on GOP voter registration efforts. The broad goal involves registering at least one million new Republican voters ahead of 2020. One specific group, Engage Texas, led by Houston real estate developer Richard Weekley, hopes to raise millions ahead of the presidential election.
“According to reports from the Washington Examiner, Engage Texas was “set up as a 501(c)4, political nonprofit organization and plans to raise and spend $25 million by Election Day next year.””
“One of the things that we face as a challenge in Texas is people assume that Texas always has been and always will be red,” Republican Party of Texas chairman James Dickey told  Sirius XM’s Breitbart News Saturday in April. “That’s not true. It was only 16 years ago that Republicans even took the Texas House.”
House Minority Leader Kevin McCarthy spoke exclusively to Breitbart News Saturday over the weekend and broke down the plan for Republicans to take back the House in 2020. He believes the Democrat Party’s mass embrace of socialism could lead to its demise in the House– at least in 2020.
“There are 31 seats that Democrats sit in today that President Trump carried,” McCarthy explained.
“Of those 31, 13 of them President Trump carried by more than six points. So, here you have the socialist wing of the party trying to take them further left, when the only way they have the majority is actually winning in Republican areas—areas that would be swing districts,” he continued.
As Breitbart News reported:
“Currently, Republicans hold 198 seats in the House. A majority in the House is 218 seats. So, if Republicans take back those 13 seats from Democrats in districts that Trump won by more than six percent, and seven more from the remaining 18 districts that Democrats represent that Trump won—while holding what they currently have—or find some other way to net gain 20 seats, they can retake the majority in 2020.”
McCarthy continued:
“Look at what the policies this new socialist Democrat party is doing. They want to one, their first most important bill—when you become the majority, you reserve the first ten numbers for your most important bills—so their most important bill, H.R. 1, was to take your taxpayer money and give it to their campaigns.” Literally, tax the American public and give them thousands of dollars for their own campaigns. Then they have what is called Medicare for None. Medicare for None would take 180 million people’s healthcare away while at the same time bankrupt those who are currently on Medicare. It’s more government control.”",
  "href": "https://www.breitbart.com/politics/2019/08/05/kenny-marchant-fourth-texas-republican-not-seek-re-election/",
  "publicationDate": 2019-08-05T10:22:49.000Z,
  "subheading": "Rep. Kenny Marchant (R-TX) announced Monday he will not seek reelection, making him the fourth Republican lawmaker to vacate his seat in the Lone Star State ahead of 2020.",
  "title": "Kenny Marchant Is Fourth Texas Republican Not to Seek Re-Election",
}
`;

exports[`breitbart snapshots does articles without subheadings 1`] = `
Object {
  "authors": Array [
    Object {
      "href": "https://www.breitbart.com/author/pam-key/",
      "name": "Pam Key",
    },
  ],
  "content": "On Friday’s broadcast of MSNBC’s “Deadline,” anchor Nicolle Wallace discussed Shepard Smith’s departure from Fox News.
Wallace said, “That’s Shep Smith in my view, the most important voice at Fox News. He took apart Donald Trump’s Uranium One scam, he was an important voice for truth in front of an audience that needs it desperately.”
She continued, “Very sad news for the truth today.”",
  "href": "https://www.breitbart.com/clips/2019/10/11/msnbcs-wallace-on-shep-smiths-fox-news-departure-very-sad-news-for-the-truth-today/",
  "publicationDate": 2019-10-11T16:07:08.000Z,
  "subheading": undefined,
  "title": "MSNBC’s Wallace on Shep Smith’s Fox News Departure: ‘Very Sad News for the Truth Today’",
}
`;

exports[`breitbart snapshots does not do ending citations 1`] = `
Object {
  "authors": Array [
    Object {
      "href": "https://www.breitbart.com/author/joel-b-pollak/",
      "name": "Joel B. Pollak",
    },
  ],
  "content": "The political officer for the U.S. embassy in Kyiv, Ukraine, was testifying in the last public hearing of the House Intelligence Committee impeachment inquiry. He said that he had overheard a phone conversation at an outdoor restaurant between President Donald Trump and U.S. Ambassador to the E.U. Gordon Sondland in which Trump had mentioned “investigations.”
Though he did not take notes of that portion of the call, and had been drinking wine, Holmes said that his memory of the conversation was clear. He shared it with colleagues at the embassy, including Charge d’affaires William Taylor — though Taylor testified last week that he had only learned about the conversation recently.
In his closed-door deposition last week, Holmes had trouble remembering how many people he had told about the conversation. He admitted, for example, boasting to six friends on vacation about hearing President Trump on the phone call.
Rep. Jim Jordan (R-OH) could hardly believe it:
“MR. JORDAN: You told friends you were sitting by an ambassador who was talking on his cell phone with the President of the United States, you told your buddies about that?
“MR. HOLMES: Yeah.
“MR. JORDAN: How many people did you teIl?
“MR. HOLMES: Sir, again, I don’t recall specifically. I don’t recall specifically. I was traveling with six friends.
“MR. JORDAN: Six fniends?
“MR. HOLMES: Yeah.
“MR. JORDAN: So now it’s up to nine people. When you come back on the 6th, who all did you tell then?
“MR. HOLMES: In the meeting with Ambassador Taylor, I told him, and the other people in that meeting would have heard it. So, as I said before, I believe it was my deputy — sir, I don’t recall who was in the room at that instance when I told them — I guess what I’m saying is that — I’m focused on telling Ambassador Taylor, he’s the person I’m there to brief. And I don’t always know who else was in the room.”
In the public hearing, Rep. Mike Conaway (R-TX) asked Holmes to commit to being more careful with presidential communications in future. Holmes refused.
The following exchange ensued:
“Conaway: Mr. Holmes, in your role, you’re privileged to an awful lot of stuff. Official things, and things that are best kept between you and the official folks that you deal with. Is there an expectation among the principals that you represent that you will exercise some discretion in what you share with others about what goes on?
“Holmes: Yes, sir.
“Conaway: In your public — in your testimony in your deposition you made the — first off, we had a hard time pinning down the number of people that you’ve actually had this conversation with about the conversation that you overheard. Now, our ambassador had no expectation of privacy, you know, blustering around, doing what he’s done, but we couldn’t figure how many people you actually shared that information with. And I would argue that the information is unflattering to the president, unflattering to the ambassador, and that your discretion is, you know, at odds here. Your testimony, your deposition said you shared that with folks who you thought would find it “interesting.” I’d argue that everybody on the back row would find it interesting, but I don’t know that that’s necessarily criterion. On a go-forward basis, can you articulate in the future when you’re privileged to certain circumstances that would be embarrassing to the principal — that if it’s official, that you share with the ambassador, fine, but that folks outside the embassy or folks even within the embassy that don’t have a need to know, that you wouldn’t regale them with your recounting of those instances?
“Holmes: Sir, I think it was Gordon Sondland who showed indiscretion having that conversation over a phone line —”
After some crosstalk, Conaway tried again:
“Conaway: The question is of you, Mr. Holmes. Your discretion. Gordon Sondland did not expect privacy. I got that. But you’ve been in a room 17 years where people trust that whatever went on in the room and left you kept it to official channels and didn’t share all that information with other folks. I’m just asking you to argue on your own behalf that “interesting” is not some sort of a criterion that you would use when you share information from meetings. A simple, straightforward question.
“Holmes: Sir, I shared the information I needed to share with the right people who needed to know it. I did not share any information that people didn’t need to know.”
Committee chair Rep. Adam Schiff (D-CA) cut Conaway off. Holmes did not clarify why his six vacation “buddies” needed to know.",
  "href": "https://www.breitbart.com/politics/2019/11/21/watch-david-holmes-refuses-to-say-he-will-be-more-careful-with-communications/",
  "publicationDate": 2019-11-21T14:25:59.000Z,
  "subheading": "State Department staffer David Holmes refused to promise Congress on Thursday afternoon that he would be more responsible in future with sensitive communications.",
  "title": "Watch: David Holmes Refuses to Say He Will Be More Careful with Communications",
}
`;

exports[`breitbart snapshots does not put the subheading in the article and correctly ignore ad 1`] = `
Object {
  "authors": Array [
    Object {
      "href": "https://www.breitbart.com/author/breitbart-london-news/",
      "name": "Breitbart London",
    },
  ],
  "content": "Friday’s decision upheld the government’s appeal against that ruling. Reasons for the decision will be published next month.
Justice Minister Ferd Grapperhaus has welcomed the decision as supporting the government’s policy of not repatriating women and children from Syria if they went there to join the Islamic State group.
Grapperhaus says “these women made the choice, with or without children, to travel to IS territory and join a terrorist organisation. The Cabinet will not actively bring them back from the region.”
Lawyers representing 23 women and their 56 children sought an injunction earlier this month forcing the government to repatriate them.",
  "href": "https://www.breitbart.com/europe/2019/11/23/court-dutch-govt-does-not-have-repatriate-children-syria/",
  "publicationDate": 2019-11-23T00:46:51.000Z,
  "subheading": "THE HAGUE, Netherlands (AP) — A Dutch appeals court has overturned a lower court’s order that the government must attempt to bring home children whose mothers travelled to Syria to join Islamic extremist groups.",
  "title": "Court: Dutch Govt Does Not Have to Repatriate Children from Syria",
}
`;

exports[`breitbart snapshots ignores twitter widgets and puts smart quotes around blockquotes 1`] = `
Object {
  "authors": Array [
    Object {
      "href": "https://www.breitbart.com/author/ildefonso-ortiz/",
      "name": "Ildefonso Ortiz",
    },
  ],
  "content": "The attack took place on Monday when three women from the Lebaron family were driving their 14 children from the town of Bavispe, Sonora, to a rural Mormon community called La Mora. According to Mexico’s SinEmbargo.com.mx, Rhonita Miller Lebaron was driving one of the vehicles with her four children–two six-month-old twins and two ages 8 and 9–when her SUV got a flat tire. Her relatives turned back towards Bavispe to bring a spare.
“Asi se vive en el Gobierno de @lopezobrador_ Mormones Mexicanos, mujeres y niños inocentes emboscados en la Sierra de Chihuahua son acribillados y quemados vivos por los Carteles que mandan en Mexico! @Javier_Corral @revistaproceso @CarlosLoret @CiroGomezL pic.twitter.com/uQNJYbeO73
“— LeBaron (@AlexLebaron1) November 4, 2019”
When the relatives returned, they found large plumes of smoke coming from the roadside scene. The relatives found a charred SUV with their loved inside the vehicle. Soon after, a group of cartel gunmen attacked and forced them to run into the brush for cover.
The two other women and their children were struck numerous times by gunfire. Six of the children ran into the brush and survived, but one young girl is still missing. In the aftermath, cartel gunmen sealed the area for hours to keep news of the massacre from spreading. Lebaron family members in the U.S. took to social media to raise awareness about the murders.
Initial reports from the family said 12 members were killed. Over night, the number of dead adjusted to 10. Alfonso Durazo, Mexico’s Public Security Secretary, announced Tuesday morning that the death toll stands at three women and six children. Another six children were injured and one more remains missing as of Tuesday morning.",
  "href": "https://www.breitbart.com/border/2019/11/05/narco-terror-10-u-s-women-children-murdered-by-cartel-gunmen-near-new-mexico-border/",
  "publicationDate": 2019-11-05T04:15:36.000Z,
  "subheading": "A group of cartel gunmen ambushed and murdered three women and six children near a rural Mormon community in Mexico, close to the New Mexico border. Six more children, all from the same family, were also injured. One additional girl remains missing. Relatives stated that all of the victims are dual U.S. and Mexican citizens.",
  "title": "NARCO-TERROR: 9 U.S. Women, Children Murdered by Cartel Gunmen near New Mexico Border",
}
`;
